ddspipe_participants/src/cpp/reader/dds/CommonReader.cpp

```diff
@@ -103,9 +104,21 @@ void CommonReader::on_data_available(
 {
     logInfo(DDSPIPE_DDS_READER, "On data available in reader in " << participant_id_ << " for topic " << topic_ << ".");
 
+    monitor_msg_rx(topic_, participant_id_);
```

Comment:
Remember this may get called for several messages.

Reply:
Yeah, and?